### PR TITLE
Tidb support

### PR DESCRIPTION
**Pull request by morgo**
Add version detection for TiDB (detected_server).
Added support for tidb_snapshot option.
Auto-set tidb_snapshot to 1 second ago when TiDB detected.
Added optimization to dump _tidb_row id when it exists.
+ added a .gitignore file.

Tested on TiDB 2.1 + MySQL 5.7.19 (to make sure detected_server matched both correctly).

### CODE REVIEW

**Comment by morgo** [comment]
Just an FYI: I removed a feature that I had added, but we no longer recommend (`_tidb_row_id`).  The diff is a little cleaner now, and what is tidb related is pretty self-contained.

File: mydumper.c [hunk 1]
```diff
@@ -2500,6 +2552,10 @@ void dump_schema_data(MYSQL *conn, char *database, char *table, char *filename)
 		if (!skip_tz) {
 			g_string_append(statement,"/*!40103 SET TIME_ZONE='+00:00' */;\n");
 		}
+	} else if (detected_server == SERVER_TYPE_TIDB) {
+		if (!skip_tz) {
+			g_string_append(statement,"/*!40103 SET TIME_ZONE='+00:00' */;\n");
```
Comment (maxbube) [inline mydumper.c]: shouldn't this be `g_string_printf` ?

File: mydumper.c [hunk 2]
```diff
@@ -2923,6 +2979,10 @@ guint64 dump_table_data(MYSQL * conn, FILE *file, char *database, char *table, c
 					if (!skip_tz) {
 					  g_string_append(statement,"/*!40103 SET TIME_ZONE='+00:00' */;\n");
 					}
+				} else if (detected_server == SERVER_TYPE_TIDB) {
+					if (!skip_tz) {
+						g_string_append(statement,"/*!40103 SET TIME_ZONE='+00:00' */;\n");
```
Comment (maxbube) [inline mydumper.c]: shouldn't this be `g_string_printf` ?

File: mydumper.c [hunk 3]
```diff
@@ -1353,6 +1375,37 @@ void start_dump(MYSQL *conn)
...
+ 
+				MYSQL_RES *result = mysql_store_result(conn);
+				MYSQL_ROW row = mysql_fetch_row(result); /* There should never be more than one row */ 
+				tidb_snapshot = g_strdup(row[1]);
+				mysql_free_result(result);
+			}
+ 
+		}
+ 
+	// Need to set the @@tidb_snapshot for the master thread
```
Comment (maxbube) [inline mydumper.c]: can you fix indentation for this block?

**Comment by morgo** [comment]
I have addressed the comments in https://github.com/maxbube/mydumper/pull/155/commits/5e612b0df60c3f107d362fc736f13fdbe4d48d9e

Thank you for the review!

**Comment by ihipop** [comment]
and update on this PR?

**Pull request merged**